pinot-segment-local/src/main/java/org/apache/pinot/segment/local/recordtransformer/NullValueTransformer.java

```diff
@@ -20,16 +20,20 @@
 
 import java.util.HashMap;
 import java.util.Map;
+import org.apache.pinot.spi.config.table.TableConfig;
 import org.apache.pinot.spi.data.FieldSpec;
 import org.apache.pinot.spi.data.FieldSpec.FieldType;
 import org.apache.pinot.spi.data.Schema;
 import org.apache.pinot.spi.data.readers.GenericRow;
+import org.apache.pinot.spi.utils.TimeConverter;
 
 
 public class NullValueTransformer implements RecordTransformer {
   private final Map<String, Object> _defaultNullValues = new HashMap<>();
+  private final TimeConverter _timeConverter;
+  private final String _timeColumnName;
 
-  public NullValueTransformer(Schema schema) {
+  public NullValueTransformer(TableConfig tableConfig, Schema schema) {
     for (FieldSpec fieldSpec : schema.getAllFieldSpecs()) {
       if (!fieldSpec.isVirtualColumn() && fieldSpec.getFieldType() != FieldType.TIME) {
```

Comment:
The time field can be of type `TIME` or `DATE_TIME`, so we might want to check the field name instead

Reply:
Looks like the existing logic will only leave out `TIME `type  as null.  `DATE_TIME` will be filled with default value by 
`org.apache.pinot.spi.data.FieldSpec.getDefaultNullValue`.

Reply:
The current code has not been updated accordingly when we deprecate the `TIME` field spec. The main time column can be defined as a `DATE_TIME` field (also this is recommended), so we should skip the main time column, instead of the `TIME` field.

Reply:
Got it. Will update this part.